ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBKeyValue.java

```diff
@@ -0,0 +1,221 @@
+	private static AmazonDynamoDB client;
+	private static DynamoDB dynamoDB;
+	private static AWSCredentialsProvider awsCredentialsProvider;
+	private DynamoDBConfigs config;
+	private static Table table;
+
+	private DataGenerator dataGenerator;
+
+	/**
+	 * Credentials will be loaded based on the environment. In AWS, the credentials
+	 * are based on the instance. In a local deployment they will have to provided.
+	 */
+	@Inject
+	public DynamoDBKeyValue(AWSCredentialsProvider credential, DynamoDBConfigs config, DataGenerator dataGenerator) {
+		this.config = config;
+		if (System.getenv(NdBenchConstants.DISCOVERY_ENV).equals("AWS")) {
+			awsCredentialsProvider = credential;
+		} else {
+			/*
+			 * The ProfileCredentialsProvider will return your [default] credential profile
+			 * by reading from the credentials file located at
+			 * (/home/username/.aws/credentials).
+			 */
+			awsCredentialsProvider = new ProfileCredentialsProvider();
+			try {
+				awsCredentialsProvider.getCredentials();
+			} catch (Exception e) {
+				throw new AmazonClientException("Cannot load the credentials from the credential profiles file. "
+						+ "Please make sure that your credentials file is at the correct "
+						+ "location (/home/<username>/.aws/credentials), and is in valid format.", e);
+			}
+		}
+	}
+
+	@Override
+	public void init(DataGenerator dataGenerator) throws Exception {
+		logger.info("Initing DynamoDB plugin");
+		client = AmazonDynamoDBClientBuilder.standard().withCredentials(awsCredentialsProvider).build();
+		dynamoDB = new DynamoDB(client);
+
+		/*
+		 * Create a table with a primary hash key named 'name', which holds a string.
+		 * Several properties such as provisioned throughput and atribute names are
+		 * defined in the configuration interface.
+		 */
+
+		logger.debug("Creating table if it does not exist yet");
+		table = dynamoDB.createTable(this.config.getTableName(),
+				Arrays.asList(new KeySchemaElement(config.getAttributeName(), KeyType.HASH)),
+				Arrays.asList(new AttributeDefinition("Id", ScalarAttributeType.N),
+						new AttributeDefinition("value", ScalarAttributeType.S)),
+				new ProvisionedThroughput(this.config.getReadCapacityUnits(), this.config.getWriteCapacityUnits()));
+
+		logger.debug("Waiting until the table is in ACTIVE state");
+		table.waitForActive();
+
+		DescribeTableRequest describeTableRequest = new DescribeTableRequest()
+				.withTableName(this.config.getTableName());
+		TableDescription tableDescription = client.describeTable(describeTableRequest).getTable();
+		logger.info("Table Description: " + tableDescription);
+
+		logger.info("DynamoDB Plugin initialized");
+	}
+
+	/**
+	 * 
+	 * @param key
+	 * @return the item
+	 * @throws Exception
+	 */
+	@Override
+	public String readSingle(String key) throws Exception {
+		Item item = null;
+		try {
+			GetItemSpec spec = new GetItemSpec().withPrimaryKey("Id", key).withConsistentRead(config.consistentRead());
```

Comment:
consistent reads will use double the RCUs and can't be served by DAX.  They're also only consistent within the region and won't be globally consistent when using cross region replication.

Reply:
Right, the idea is to test it in comparison with a quorum Cassandra read. By default, we will not be using it when we want to test DAX: `@DefaultValue("false")`.